src/bigtable/mod.rs

```diff
@@ -0,0 +1,537 @@
+                entry.status.as_ref().map(|status| status.code != 0) == Some(true)
+            }));
+        }
+
+        if !errors.is_empty() {
+            Err(errors.into())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Performs a mutation request for a single row.
+    pub async fn mutate_row(&mut self, request: MutateRowRequest) -> Result<(), tonic::Status> {
+        let mut retry = self.retry.new_operation();
+
+        while let Err(e) = self.inner.mutate_row(request.clone()).await {
+            if let Some(sleep) = retry.check_retry(&(), &e) {
+                sleep.await;
+            } else {
+                return Err(e);
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Delete one or more rows from the table.
+    pub async fn delete_rows(
+        &mut self,
+        table_name: &str,
+        row_keys: impl IntoIterator<Item = impl Into<Bytes>>,
+    ) -> Result<(), MutateRowsError> {
+        let table_name = format!("{}{}", self.table_prefix, table_name);
+        let req =
+            MutateRowsRequest::new(table_name).with_entries(row_keys.into_iter().map(|row_key| {
+                mutation::Entry::new(row_key.into()).with_mutation(mutation::DeleteFromRow {})
+            }));
+        self.mutate_rows(req).await
+    }
+
+    /// Set data for a collection of rows, all having the same table name and column family name.
+    pub async fn set_rows_data<RowKey, RowData, ColName, CellData>(
+        &mut self,
+        table_name: &str,
+        family_name: String,
+        data: impl IntoIterator<Item = (RowKey, RowData)>,
+    ) -> Result<(), MutateRowsError>
+    where
+        RowKey: Into<Bytes>,
+        ColName: Into<Bytes>,
+        CellData: Into<Bytes>,
+        RowData: IntoIterator<Item = (ColName, CellData)>,
+    {
+        self.set_rows_data_with_timestamp(table_name, family_name, -1, data)
+            .await
+    }
+
+    /// Set some data for a single row.
+    pub async fn set_row_data<RowKey, RowData, ColName, CellData>(
+        &mut self,
+        table_name: &str,
+        family_name: String,
+        row_key: RowKey,
+        data: RowData,
+    ) -> Result<(), tonic::Status>
+    where
+        RowKey: Into<Bytes>,
+        ColName: Into<Bytes>,
+        CellData: Into<Bytes>,
+        RowData: IntoIterator<Item = (ColName, CellData)>,
+    {
+        self.set_row_data_with_timestamp(table_name, family_name, -1, row_key, data)
+            .await
+    }
+
+    /// Set data for a collection of rows, all having the same table name and column family name.
+    ///
+    /// For more on the purpose of the `timestamp` field, see
+    /// [Google's docs](https://cloud.google.com/bigtable/docs/garbage-collection#timestamps).
+    pub async fn set_rows_data_with_timestamp<RowKey, RowData, ColName, CellData>(
+        &mut self,
+        table_name: &str,
+        family_name: String,
+        timestamp: i64,
```

Comment:
similar to `rows_limit`, what do you think about `timestamp: Option<u64>` which maps None to -1 internally?